go-controller/pkg/ovn/default_network_controller.go

```diff
@@ -906,22 +925,25 @@ func (h *defaultNetworkControllerEventHandler) UpdateResource(oldObj, newObj int
 					nodeSubnetChanged(oldNode, newNode) || hostAddressesChanged(oldNode, newNode) ||
 					nodeGatewayMTUSupportChanged(oldNode, newNode))
 				_, hoSync := h.oc.hybridOverlayFailed.Load(newNode.Name)
+				_, syncZoneIc := h.oc.syncZoneIcFailed.Load(newNode.Name)
 				nodeSyncsParam = &nodeSyncs{
 					nodeSync,
 					clusterRtrSync,
 					mgmtSync,
 					gwSync,
-					hoSync}
+					hoSync,
+					syncZoneIc}
 			} else {
 				klog.Infof("Node %s moved from the remote zone %s to local zone.",
 					newNode.Name, util.GetNodeZone(oldNode), util.GetNodeZone(newNode))
 				// The node is now a local zone node.  Trigger a full node sync.
-				nodeSyncsParam = &nodeSyncs{true, true, true, true, true}
+				nodeSyncsParam = &nodeSyncs{true, true, true, true, true, config.OVNKubernetesFeature.EnableInterconnect}
 			}
 
 			return h.oc.addUpdateLocalNodeEvent(newNode, nodeSyncsParam)
 		} else {
-			return h.oc.addUpdateRemoteNodeEvent(newNode)
+			_, syncZoneIc := h.oc.syncZoneIcFailed.Load(newNode.Name)
+			return h.oc.addUpdateRemoteNodeEvent(newNode, syncZoneIc)
```

Comment:
so this might actually be called for every node update? one of the main reasons we added the whole nodeSyncParam was to be careful what updates of a node object we care about and what we don't care about... we really don't want to be calling this logic which in turn calls the `zoneChassisHandler.AddRemoteZoneNode` and `zoneICHandler.AddRemoteZoneNode` everytime there is a node update...

Reply:
addUpdateLocalNodeEvent will be called for every node update too right ?   I guess addUpdateLocalNodeEvent() will check the nodesync params will call the appropriate functions.   If all the nodeSync params are false,  I guess it would be a no-op.

Same thing here,  addUpdateRemoteNodeEvent() will calls the zoneChassisHandler and zoneICHandler only of syncZoneIC is true.